vmm/src/device_manager.rs

```diff
@@ -1712,8 +1735,8 @@ impl DeviceManager {
             id.clone()
         } else {
             let id = self.next_device_name(VFIO_DEVICE_NAME_PREFIX)?;
-            device_cfg.id = Some(id.clone());
-            id
+            device_cfg.id = id.clone();
+            id.unwrap()
```

Comment:
unwrap()?

Reply:
@sboeuf Yes the previously the function returned a `String` which was then wrapped into a `Some` and saved with the plain `String` returned. Now it returns an `Option<String>` which is saved directly but the `String` needs to be returned so it needs to be unwrapped.

Reply:
We don't have to worry about `None` here as on the `feature = "pci_support" case the function either errors or returns `Some(..)`